near-contract-standards/src/non_fungible_token/payout/payout_impl.rs

```diff
@@ -0,0 +1,80 @@
+use super::NonFungibleTokenPayout;
+use crate::non_fungible_token::core::NonFungibleTokenCore;
+use crate::non_fungible_token::payout::*;
+use crate::non_fungible_token::NonFungibleToken;
+use near_sdk::assert_one_yocto;
+use near_sdk::{require, AccountId, Balance, IntoStorageKey};
+use std::collections::HashMap;
+impl Royalties {
+    /*   pub fn new(accounts: HashMap<AccountId, u8>, percent: u8) -> Self {
+        let this = Self { accounts, percent };
+        this.validate();
+        this
+    } */
+    pub fn new<S>(key_prefix: S) -> Self
+    where
+        S: IntoStorageKey,
+    {
+        let temp_accounts: HashMap<AccountId, u8> = HashMap::new();
+        let this =
+            Self { key_prefix: key_prefix.into_storage_key(), accounts: temp_accounts, percent: 0 };
+        this.validate();
+        this
+    }
+    pub(crate) fn validate(&self) {
+        require!(self.percent <= 100, "royalty percent must be between 0 - 100");
+        require!(
+            self.accounts.len() <= 10,
+            "can only have a maximum of 10 accounts spliting royalties"
+        );
+        let mut total: u8 = 0;
+        self.accounts.iter().for_each(|(_, percent)| {
+            require!(*percent <= 100, "each royalty should be less than 100");
+            total += percent;
+        });
+        require!(total <= 100, "total percent of each royalty split  must be less than 100")
+    }
+    pub fn create_payout(&self, balance: Balance, owner_id: &AccountId) -> Payout {
+        let royalty_payment = apply_percent(self.percent, balance);
+        let mut payout = Payout {
+            payout: self
+                .accounts
+                .iter()
+                .map(|(account, percent)| {
+                    (account.clone(), apply_percent(*percent, royalty_payment).into())
+                })
+                .collect(),
+        };
+        let rest = balance - royalty_payment;
+        let owner_payout: u128 = payout.payout.get(owner_id).map_or(0, |x| x.0) + rest;
+        payout.payout.insert(owner_id.clone(), owner_payout.into());
+        payout
+    }
+}
+
+fn apply_percent(percent: u8, int: u128) -> u128 {
+    int * percent as u128 / 100u128
```

Comment:
Seems like this can overflow, are there any guarantees around this? Probably want to do a checked multiplication with an `expect` to make sure someone doesn't need overflow-checks enabled to catch this